packages/venia-concept/src/RootComponents/Category/category.js

```diff
@@ -1,119 +1,97 @@
+    const { runQuery, setLoading } = queryApi;
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-        const pageControl = {
-            currentPage: currentPage,
-            setPage: setCurrentPage,
-            updateTotalPages: setPrevPageTotal,
-            totalPages: prevPageTotal
-        };
+    useEffect(() => {
+        setLoading(true);
+        runQuery({
+            variables: {
+                id: Number(id),
+                onServer: false,
+                pageSize: Number(pageSize),
+                currentPage: Number(currentPage)
+            }
+        });
+    }, [id, pageSize, currentPage, runQuery, setLoading]);
 
-        return (
-            <Query
-                query={categoryQuery}
-                variables={{
-                    id: Number(id),
-                    onServer: false,
-                    pageSize: Number(pageSize),
-                    currentPage: Number(currentPage)
-                }}
-            >
-                {({ loading, error, data }) => {
-                    if (error) return <div>Data Fetch Error</div>;
-                    // If our pagination component has mounted, then we have
-                    // a total page count in the store, so we continue to render
-                    // with our last known total
-                    if (loading)
-                        return pageControl.totalPages ? (
-                            <CategoryContent
-                                pageControl={pageControl}
-                                pageSize={pageSize}
-                            />
-                        ) : (
-                            loadingIndicator
-                        );
+    useEffect(() => {
+        window.scrollTo({
+            left: 0,
+            top: 0,
+            behavior: 'smooth'
+        });
+    }, [currentPage]);
 
-                    // TODO: Retrieve the page total from GraphQL when ready
-                    const pageCount =
-                        data.category.products.total_count / pageSize;
-                    const totalPages = Math.ceil(pageCount);
-                    const totalWrapper = {
-                        ...pageControl,
-                        totalPages: totalPages
-                    };
-
-                    return (
-                        <CategoryContent
-                            classes={classes}
-                            pageControl={totalWrapper}
-                            data={data}
-                        />
-                    );
-                }}
-            </Query>
+    if (error) return <div>Data Fetch Error</div>;
+    // If our pagination component has mounted, then we have
+    // a total page count in the store, so we continue to render
+    // with our last known total
+    if (loading || !data)
+        return pageControl.totalPages ? (
+            <CategoryContent pageControl={pageControl} pageSize={pageSize} />
+        ) : (
+            loadingIndicator
```

Comment:
I don't think we want to show a loading indicator when we're not loading. If a request was not initiated, this would be loading forever. Consider returning `null` or something when we're in that empty state.